pkg/util/log/test_log_scope.go

```diff
@@ -44,26 +41,19 @@ type tShim interface {
 	Failed() bool
 	Error(...interface{})
 	Errorf(fmt string, args ...interface{})
+	Name() string
 }
 
 var showLogs bool
 
-// Scope creates a TestLogScope which corresponds to the lifetime of a
-// logging directory. If testName is empty, the logging directory is
-// named after the caller of Scope, up `skip` caller levels. It also
+// Scope creates a TestLogScope which corresponds to the lifetime of a logging
+// directory. The logging directory is named after the calling test. It also
 // disables logging to stderr for severity levels below ERROR.
-func Scope(t tShim, testName string) *TestLogScope {
+func Scope(t tShim) *TestLogScope {
 	if showLogs {
 		return (*TestLogScope)(nil)
 	}
-	if testName == "" {
-		testName = "logUnknown"
-		if _, _, f := caller.Lookup(1); f != "" {
-			parts := strings.Split(f, ".")
-			testName = "log" + parts[len(parts)-1]
-		}
-	}
-	tempDir, err := ioutil.TempDir("", testName)
+	tempDir, err := ioutil.TempDir("", t.Name())
```

Comment:
I think this will not work in cases when test have subtests. I tried doing this as part of my PR as well. LGTM conceptually though.